pkg/requests/base.go

```diff
@@ -488,3 +499,35 @@ func normalizePath(path string) string {
 
 	return "/v1/" + path
 }
+
+func (rb *Base) experimentalRequestSigning(req *http.Request, experimentalFields config.ExperimentalFields) error {
+	privKey := experimentalFields.PrivateKey
+
+	keyToValues := strings.Split(strings.Trim(experimentalFields.StripeHeaders, ";"), ";")
+	for _, pair := range keyToValues {
+		header := strings.Split(pair, "=")
+		if len(header) != 2 {
+			continue
+		}
+		headerName := header[0]
+		headerValue := header[1]
+		if headerName == stripeContextHeaderName {
+			displayMessage := fmt.Sprintf("Operating in %s %s\n", ansi.Bold(experimentalFields.ContextualName), ansi.Color(os.Stdout).Gray(10, "("+headerValue+")..."))
+			fmt.Print(ansi.Color(os.Stdout).Gray(10, displayMessage))
+		} else if headerName == authorizationHeaderName && privKey == "" {
+			creds, err := rb.Profile.GetSessionCredentials()
```

Comment:
Is it necessary to write the private key to the config file when we can we read them from the credential manager? I have a couple concerns:
1. Duplicating the field between the config file and credential manager could lead to them going out of sync.
2. I'm not really comfortable showing the private key in cleartext, since this could limit what we can do security-wise. For example, if we wanted to extend the session expiration from ~hours to ~days, this could make it difficult to do in a secure way.

If we can get away with not having writing this in the config file, I think we should do that.

Reply:
it's not necessary specifically for this case. I wasn't sure if there were edge cases where the private key is only available in the config file, which is why this code is there. But if we can reliably get it for all OSes from `GetSessionCredentials`, then im fine removing private key from the experimental section

Reply:
I'm fairly confident `GetSessionCredentials` will be okay for mac and windows, but I haven't been able to validate this on linux. Given that this change is experimental still, I think this is okay but adding a ticket for validating on all OSes could be useful.